lib/core.js

```diff
@@ -852,7 +864,8 @@ function parseHeader(info) {
     tree: info.head || getDefaultTree(),
     hints: {
       reorgs: [],
-      contiguousLength: info.hints ? info.hints.contiguousLength : 0
+      contiguousLength: info.hints ? info.hints.contiguousLength : 0,
+      remoteContiguousLength: info.hints ? info.hints.remoteContiguousLength : 0
```

Comment:
Might be irrelevant in practice, but in theory `info.hints` can be set but without both of these fields present, since they're optional, so something like `remoteContiguousLength: info.hints?.remoteContiguousLength ? info.hints.remoteContiguousLength : 0` seems safer

Reply:
Good point. Especially the `remoteContiguousLength` could be undefined if used with an older `hypercore-storage` version.

I updated and also used nullish coalescing to make it easier to read.